### PR TITLE
Add successful layers parameter to the buildFailure hook

### PR DESCRIPTION
**Pull request by CameronDiver**
It may be useful for a cloud service to store all of the layers which
did succeed when a build fails, and before this commit that was not
possible without doing some log parsing.

This commit adds a `layers` paramter to the buildFailure hook, which is
all of the successful layers.

Also added tests.

Change-type: minor
Signed-off-by: Cameron Diver <cameron@resin.io>

### CODE REVIEW

File: src/builder.ts [hunk 1]
```diff
@@ -100,10 +100,14 @@ export default class Builder {
 
 			// Catch any errors the stream produces
 			outputStream.on('error', (err: Error) => {
-				self.callHook(hooks, 'buildFailure', handler, err)
+				let layers = self.layers
+				layers.pop()
```
Comment (flesler) [inline src/builder.ts]: Worth mentioning that `pop()` modifies the original array and I see there are several places where `pop()` is called on the same object and it's then passed around. Are you sure it should be dropping layers on all those cases? Just double checking.

Comment (CameronDiver) [inline src/builder.ts]: The times that they are called are mutually exclusive, but I take your point, probably would be best to copy.

File: src/builder.ts [hunk 2]
```diff
@@ -100,10 +100,14 @@ export default class Builder {
 
 			// Catch any errors the stream produces
 			outputStream.on('error', (err: Error) => {
-				self.callHook(hooks, 'buildFailure', handler, err)
+				const layers = self.layers.slice()
```
Comment (flesler) [inline src/builder.ts]: Better:
`const layers = self.layers.slice(0, -1)`

Comment (CameronDiver) [inline src/builder.ts]: good point :+1:

**Pull request merged**